hazelcast/src/main/java/com/hazelcast/query/impl/IndexStore.java

```diff
@@ -187,16 +194,25 @@ Iterator<QueryableEntry> getSqlRecordIterator(Comparable from, boolean fromInclu
     Iterator<IndexKeyEntries> getSqlRecordIteratorBatch(boolean descending);
 
     /**
+     * Returns records matching comparison. There are no records {@code < NULL}.
+     * Trying to get records {@code <= NULL} is invalid. However, you can get records {@code > NULL}
```

Comment:
To be honest, ` records {@code < NULL}` and `records {@code <= NULL}` and are a bit confusing and hard to understand even after few re-readings. Can you re-word this doc, please?

Reply:
https://github.com/hazelcast/hazelcast/pull/25328/commits/9fb7383f38c58bc3aa4ae2e0c133d00e8bf0edcd

IMO this methods contract is hard to comprehend, but that is how it works. I only described it. These methods could be simplified but I do not understand why do we have so many variants of them.